test/unit/query-generator/create-database-query.test.ts

```diff
@@ -0,0 +1,61 @@
+import { buildInvalidOptionReceivedError } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/check.js';
+import { expectsql, getTestDialect, sequelize } from '../../support';
+
+const dialectName = getTestDialect();
+
+const notSupportedError = new Error(`Databases are not supported in ${dialectName}.`);
+
+describe('QueryGenerator#createDatabaseQuery', () => {
```

Comment:
Snowflake had variants on these tests for when `quoteIdentifiers` is false. We should add those as well right?
Update; I see you do have it in the tests for `QueryGenerator#dropDatabaseQuery`

Reply:
`quoteIdentifiers` is handled by the same quoteIdentifier method every time, so I don't think it's necessary to test all of them. That being said `quoteIdentifiers` is an option I don't understand the purpose of. We need the table names to be quoted when they have characters that could be interpreted differently, such as differences in casing. I don't see the point of allowing to disable it

Reply:
That makes sense. Shall we disable that in another PR?

Reply:
Yes we'll investigate separately from this one